table.csv

```diff
@@ -447,3 +447,7 @@ leofcoin-pr,          Leofcoin Peer Reputation,                         0x83
 torrent-info,         Torrent file info field (bencoded),               0x7b
 torrent-file,         Torrent file (bencoded),                          0x7c
 ed25519-pub,          Ed25519 public key,                               0xed
+
+CIDs,,
```

Comment:
Maybe "content namespaces"?. We're not really defining new types of CIDs.

1. ipld-ns: <ipld-ns-codec><cid>[/arbitrary/path]
2. ipfs-ns: <ipld-ns-codec><cid>[/arbitrary/path]
3. swarm-ns: <swarm-ns-codec><cid>[byte range?]

We could also merge #93 to allow arbitrary string paths (although we don't *have* to do it that way).

Basically, I'd like to make this as flexible as possible.

Reply:
Can you elaborate on what an IPLD NS would encode, as opposed to an IPFS-NS? Would it be a CID, or some other data format?

Reply:
I'm just taking string paths and encoding them in a compact binary form.

* `/ipld/SomeStringCID/some/remaining/path` -> `<ipld-namespace-multicodec><binary-cid><path>`
* `/ipfs/SomeStringCID/some/remaining/path` -> `<ipfs-namespace-multicodec><binary-cid><path>`

So these will both be CIDs + Paths.

We need different namespaces because IPFS is built on top of IPLD. In IPFS, we' interpret the content as files and directories and the path a traversal through a directory tree. In IPLD we interpret the content as structured data and the path as a traversal through a JSON-like datastructure.

---

For the swarm namespace, I wrote `[byte range?]` as you *may* want to allow for a byte range in addition to the CID.

Reply:
As a string although we may want to length-prefix these strings.

Regardless, we can probably skip that for now. The important part here is to reserve these numbers. Regardless of what we say now, we'll probably change our minds when we actually try to prototype these.